packages/components/src/Combobox/Combobox.tsx

```diff
@@ -1,193 +1,84 @@
-  const [searchValue, setSearchValue] = useState<string>("");
-  const [selectedOption, setSelectedOption] = useState<string>("");
+function useComboboxValidation(children: ReactNode): {
+  triggerElement: ReactNode;
+  contentElement: ReactNode;
+} {
+  const childrenArray = React.Children.toArray(children);
+  let triggerElement: ReactNode,
+    contentElement: ReactNode,
+    multipleTriggersFound = false;
+
+  childrenArray.forEach(child => {
+    if (isTriggerElement(child)) {
+      if (triggerElement) {
+        multipleTriggersFound = true;
+      }
+      triggerElement = child;
+    }
+    if (isContentElement(child)) {
+      contentElement = child;
+    }
+  });
 
-  const filteredOptions = props.options.filter(option =>
-    option.toLowerCase().includes(searchValue.toLowerCase()),
+  useAssert(multipleTriggersFound, COMBOBOX_TRIGGER_COUNT_ERROR_MESSAGE);
+
+  useAssert(
+    !triggerElement || !contentElement,
+    COMBOBOX_REQUIRED_CHILDREN_ERROR_MESSAGE,
   );
 
-  function handleSearch(event: React.ChangeEvent<HTMLInputElement>) {
-    setSearchValue(event.target.value);
-  }
+  return {
+    contentElement,
+    triggerElement,
+  };
+}
 
-  function handleSelection(event: React.MouseEvent<HTMLButtonElement>) {
-    const selected = event.currentTarget.innerText;
-    setSelectedOption(selected);
-    setOpen(false);
-    setSearchValue("");
-    props.onSelection(selected);
-  }
+function isTriggerElement(child: ReactNode): boolean {
   return (
-    <div className={classnames(styles.content, !open && styles.hidden)}>
-      <div className={styles.search}>
-        <input
-          type="search"
-          className={styles.searchInput}
-          placeholder="Search teammates"
-          onChange={(event: React.ChangeEvent<HTMLInputElement>) =>
-            handleSearch(event)
-          }
-          value={searchValue}
-        />
-
-        {/* if there is a search value show the clear icon */}
-        {searchValue && (
-          <div
-            className={styles.triggerIcon}
-            onClick={() => setSearchValue("")}
-            role="button"
-          >
-            <Icon name="remove" size="small" />
-          </div>
-        )}
-      </div>
-
-      {filteredOptions.map(option => (
-        <button
-          className={classnames(
-            option === selectedOption && styles.selectedOption,
-          )}
-          role="button"
-          tabIndex={0}
-          key={option}
-          onClick={handleSelection}
-        >
-          {option}
-        </button>
-      ))}
-
-      {/* if there are no results matching the search value show no results item */}
-      {filteredOptions.length === 0 && <p>No results for {searchValue}</p>}
-
-      {/* only render actions if their are any */}
-      {props.children && <div className={styles.actions}>{props.children}</div>}
-    </div>
+    React.isValidElement(child) &&
+    typeof child !== "string" &&
+    typeof child.type !== "string" &&
```

Comment:
still had to check if it's a string eh?

Reply:
ugh, no we definitely do not now that we've said it's either one or more `ReactElement`s

that's a quick fix tho